test.py

```diff
@@ -416,9 +416,72 @@ class IntegerQueueArray(QueueArray[int]):
         self.assertRaises(Exception, test_qa.enqueue, 10)
         
     def test_queue_double_stack(self):
-        test_ds = StoreLineDoubleStack()
+        test_ds = StoreLineDoubleStack(size=1_000_000)
         self.helper(test_ds)
 
+    def test_linked_list_queue_jz(self):
+
+        class IntegerQueueLinkedList(QueueLinkedList[int]):
+            pass
+
+        queue = IntegerQueueLinkedList()
+        self.assertRaises(Exception, queue.peek)
+        self.assertRaises(Exception, queue.dequeue)
+
+        for i in range(50):
+            queue.enqueue(i)
+            self.assertEqual(queue.peek(), 0)
+            self.assertEqual(queue.size, i + 1)
+
+        for i in range(50):
+            self.assertEqual(queue.dequeue(), i)
+            self.assertEqual(queue.size, 49 - i)
```

Comment:
Same things here. You can only use functions since internally, the implementer doesn't have to use the same `size`. If there was a function like `.size()` then you could use it